zenoh-kotlin/src/commonMain/kotlin/io/zenoh/Config.kt

```diff
@@ -51,5 +57,316 @@ class Config private constructor(internal val path: Path? = null) {
         fun from(path: Path): Config {
             return Config(path)
         }
+
+        fun from(configData: ConfigData): Config {
+            val jsonConfig = Json.encodeToJsonElement(configData)
+            return Config(jsonConfig)
+        }
     }
+
+    constructor(jsonConfig: JsonElement) : this(null, jsonConfig = jsonConfig)
 }
+
+@Serializable
+data class ConfigData(
```

Comment:
Doesn't this need to be updated every time we modify the Zenoh config?

Can be a burden to maintain.
It would be better to have an abstract JSON  (if possible) I think.
Thus we do not need to check this class at every update of the config.

I'm thinking of cases like:
https://github.com/eclipse-zenoh/zenoh/pull/585
https://github.com/eclipse-zenoh/zenoh/pull/577
https://github.com/eclipse-zenoh/zenoh/pull/583

That had/will have an impact on configuration and thus on this class.

Reply:
You are right, removed.